### ISSUE TITLE
Chat BOT do site ainda em andamento

### ISSUE THREAD

**Issue by Brunn0B** [Brunn0B/Projeto-Site-Power-BI#3]
<!DOCTYPE html>
<html>

<head>
    <title>Bot de Bate-Papo</title>
    <script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs"></script>
    <script src="https://cdn.jsdelivr.net/npm/@tensorflow-models/universal-sentence-encoder"></script>

    <style>
        /* Estilo fornecido */
        @import url('https://fonts.googleapis.com/css2?family=Fredericka+the+Great&family=Monoton&family=Orbitron:wght@400;500;600;700;800;900&family=Russo+One&display=swap');

        * {
            box-sizing: border-box;
        }

        body {
            font-family: 'Orbitron', sans-serif;
            margin: 0;
            padding: 0;
            background-image: linear-gradient(to right bottom, #3c1c1c, #2b2b2b, #1c1c1c, #0f0f0f, #000000);
            color: #fff;
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100vh;
            background-color: white;
        }

        #chatbox {
            border: 1px solid #ccc;
            border-radius: 5px;
            background-color: #2a3a85;
            padding: 10px;
            max-height: 300px;
            overflow-y: auto;
            width: 300px;
            display: none;
            /* Hidden by default */
            position: fixed;
            top: 20px;
            right: 20px;
            z-index: 999;
        }

        /* Limitar largura das mensagens do chatbox */
        #chatbox p {
            word-wrap: break-word;
            max-width: 100%;
        }

        #userInput {
            width: 100%;
            padding: 8px;
            font-size: 16px;
            border: 1px solid #ccc;
            border-radius: 5px;
            margin-top: 10px;
            background-color: #454242;
            color: #fff;
        }

        p {
            margin: 5px 0;
        }

        strong {
            color: #007bff;
        }

        .user-message {
            text-align: right;
            color: #fff;
        }

        .bot-message {
            text-align: left;
            color: #fff;
        }

        h1 {
            text-align: center;
            color: #007bff;
            margin-bottom: 20px;
        }

        /* Styling for the chatbot open/close button */
        #chatbox-toggle {
            position: fixed;
            top: 20px;
            right: 20px;
            background-color: #007bff;
            color: #fff;
            padding: 10px 20px;
            border: none;
            border-radius: 5px;
            cursor: pointer;
            z-index: 999;
        }

        #chatbox-toggle.close {
            background-color: #ff0000;
        }

        #chatbox-input {
            position: fixed;
            bottom: 0;
            left: 0;
            width: 100%;
            background-color: #1c1c1c;
            padding: 10px;
            z-index: 999;
        }
    </style>

</head>

<body>
    <h1>ChatBot</h1>
    <button id="chatbox-toggle">Abrir Chat</button>
    <div id="chatbox">
        <p class="bot-message">Bem-vindo! Como posso ajudar?</p>
    </div>
    <div id="chatbox-input">
        <input type="text" id="userInput" placeholder="Digite sua mensagem aqui..." onkeyup="handleUserInput(event)">
    </div>
    <script>
        const chatbox = document.getElementById("chatbox");
        const chatboxToggle = document.getElementById("chatbox-toggle");
        const chatboxInput = document.getElementById("chatbox-input");
        const userInput = document.getElementById("userInput");
        let model;
        let chatboxOpen = false;

        // Carregar o modelo do Universal Sentence Encoder
        async function loadModel() {
            model = await tf.loadLayersModel('https://storage.googleapis.com/tfjs-models/tfjs/universal-sentence-encoder/tfjs_model/model.json');
        }
        loadModel();

        function appendMessage(sender, message) {
            const messageElement = document.createElement("p");
            messageElement.textContent = `${sender}: ${message}`;
            if (sender === "Você") {
                messageElement.classList.add("user-message");
            } else {
                messageElement.classList.add("bot-message");
            }
            chatbox.appendChild(messageElement);
        }

        async function findMostSimilarQuestion(userMessage, questions) {
            const userEmbedding = await model.embed([userMessage]);
            const similarities = {};

            for (const key of Object.keys(questions)) {
                const questionEmbedding = await model.embed([key]);
                const similarity = await tf.losses.cosineDistance(userEmbedding, questionEmbedding, 0).data();
                similarities[key] = similarity[0];
            }

            let mostSimilarQuestion = Object.keys(similarities).reduce((a, b) => similarities[a] > similarities[b] ? a : b);

            if (similarities[mostSimilarQuestion] < 0.6) {
                mostSimilarQuestion = "Desculpe, não entendi. Pode repetir?";
            }

            return mostSimilarQuestion;
        }

        function toggleChatbox() {
            chatboxOpen = !chatboxOpen;
            chatbox.style.display = chatboxOpen ? "block" : "none";
            chatboxInput.style.display = chatboxOpen ? "block" : "none";
            chatboxToggle.textContent = chatboxOpen ? "Fechar Chat" : "Abrir Chat";
            chatboxToggle.classList.toggle("close", chatboxOpen);
            if (chatboxOpen) {
                userInput.focus();
            }
        }

        chatboxToggle.addEventListener("click", toggleChatbox);

        async function handleUserInput(event) {
            if (event.key === "Enter") {
                const userMessage = userInput.value.trim().toLowerCase();
                userInput.value = "";

                appendMessage("Você", userMessage);
                // Lógica para respostas do bot
                const questions = {
                    "oi": "Olá! Como posso ajudar?",
                    "olá": "Oi! Tudo bem?",
                    "como você está?": "Estou bem, obrigado por perguntar!",
                    "qual é o seu nome?": "Eu sou um bot de exemplo.",
                    "adeus": "Até logo!",
                    "ajuda": "Claro! Estou aqui para ajudar. O que você precisa?",
                    "quem criou você?": "Fui criado pela OpenAI utilizando a tecnologia GPT-3.5.",
                    "o que você pode fazer?": "Eu posso responder a algumas perguntas e bater um papo. Tente me perguntar algo!",
                    "obrigado": "De nada! Se precisar de mais ajuda, é só perguntar.",
                    "o que é um advogado?": "Um advogado é um profissional legalmente habilitado para representar e defender os interesses de seus clientes em questões jurídicas.",
                    "quais são as áreas do direito?": "As áreas do direito abrangem diversas especialidades, como direito civil, penal, trabalhista, tributário, previdenciário, ambiental, empresarial, internacional, entre outras.",
                    "como contratar um advogado?": "Para contratar um advogado, é recomendado pesquisar e escolher um profissional especializado na área em questão, verificar sua experiência e credenciais. Também é importante discutir as taxas e honorários advocatícios antes de formalizar o contrato.",
                    "o que é uma procuração?": "Uma procuração é um documento legal que autoriza uma pessoa (o procurador) a agir em nome de outra (o outorgante) em assuntos legais ou administrativos. Existem diferentes tipos de procuração, como a procuração simples e a procuração pública.",
                    "quais são os direitos do consumidor?": "Os direitos do consumidor incluem o direito à informação, à segurança, à escolha, à reparação de danos, à desistência de compra, entre outros. Esses direitos estão previstos no Código de Defesa do Consumidor (CDC).",
                    "o que é um contrato de trabalho?": "Um contrato de trabalho é um acordo legal entre empregador e empregado que estabelece os termos e condições do emprego, como salário, jornada de trabalho, benefícios, férias, entre outros. Ele pode ser escrito ou verbal, mas é sempre recomendado que seja feito por escrito.",
                    "o que é um inventário?": "O inventário é um procedimento legal para a partilha dos bens de uma pessoa falecida entre os herdeiros. Esse processo é necessário após um falecimento para regularizar a sucessão e transferir a propriedade dos bens para os herdeiros.",
                    "o que é uma ação judicial?": "Uma ação judicial é um processo legal iniciado por uma parte (autor ou requerente) para resolver um conflito perante um tribunal. A parte contrária é chamada de réu ou requerido. As ações podem ser de diversas naturezas, como ações cíveis, trabalhistas, penais, entre outras.",
                    "o que é uma audiência?": "Uma audiência é uma sessão em que as partes envolvidas em um processo apresentam suas argumentações perante o juiz. É um momento importante para a instrução do processo e para que as partes apresentem suas provas e argumentos.",
                    "como funciona o divórcio?": "O divórcio é a dissolução do casamento e pode ser feito de forma consensual ou litigiosa, dependendo da concordância das partes. É um procedimento jurídico que envolve questões como partilha de bens, guarda e visitação dos filhos, pensão alimentícia, entre outros.",
                    "o que é a advocacia preventiva?": "A advocacia preventiva consiste em assessorar e orientar clientes para evitar litígios e problemas legais futuros. O advogado atua de forma preventiva, analisando contratos, resolvendo conflitos antes que se agravem e fornecendo orientações para evitar problemas jurídicos.",
                    "o que é o Tribunal de Justiça?": "O Tribunal de Justiça é o órgão máximo do Poder Judiciário estadual. É responsável por julgar recursos contra decisões proferidas pelos juízes de primeira instância e também possui competência para julgar algumas matérias de forma originária.",
                    "o que é o Ministério Público?": "O Ministério Público é uma instituição pública que atua na defesa dos interesses da sociedade e na fiscalização do cumprimento das leis. O MP pode propor ações judiciais, investigar crimes e atuar em defesa dos direitos coletivos.",
                    "o que é uma sociedade de advogados?": "Uma sociedade de advogados é um escritório composto por dois ou mais advogados que se associam para atuar em conjunto e prestar serviços jurídicos. A sociedade de advogados permite a união de esforços e a especialização em diferentes áreas do direito.",
                    "quais são as atribuições da OAB?": "A Ordem dos Advogados do Brasil (OAB) é uma instituição que representa a classe dos advogados. Suas atribuições incluem a fiscalização do exercício profissional, a defesa das prerrogativas dos advogados, a promoção de cursos e eventos jurídicos, entre outras.",
                    "o que é o Estatuto da Advocacia e da OAB?": "O Estatuto da Advocacia e da OAB é a lei que regulamenta o exercício da advocacia no Brasil. Ele estabelece os direitos e deveres dos advogados, as normas de ética profissional, as regras para inscrição na OAB, entre outros aspectos importantes para a atuação dos advogados."
                };
                let botResponse = questions[userMessage] || await findMostSimilarQuestion(userMessage, questions);

                setTimeout(() => {
                    appendMessage("Bot", botResponse);
                    chatbox.scrollTop = chatbox.scrollHeight; // Rolar para a parte inferior do chatbox
                }, 500);
            }
        }
    </script>
</body>

</html>

**Comment by Brunn0B** [comment]
Este é um bot para Trazer auxilio localizado No botão "Fale conosco" Assim possibilita uma melhora na assistência do site

**Comment by Brunn0B** [comment]
Chat bot atualizado foram adicionados diversos recursos e um parte visual minimamente agradavel
